### ISSUE TITLE
Fn::Join in Fn::Sub

### ISSUE THREAD

**Issue by rschick** [monken/cfn-include#25]
I've been having problems with doing this style of include for UserData:

```yaml
UserData:
  Fn::Base64:
    Fn::Sub:
      !Include { type: literal, location: userdata.sh }
```

It fails validation with:

```
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template error: One or more Fn::Sub intrinsic functions don't specify expected arguments. Specify a string as first argument, and an optional second argument to specify a mapping of values to replace in the string
```

It looks like an issue with CF. The output looks fine:

```json
        "UserData": {
          "Fn::Base64": {
            "Fn::Sub": {
              "Fn::Join": [
                "",
                [
                  "#!/bin/bash -xe\n",
                 ...
                ]
              ]
            }
          }
        }
```

I am currently working around this by including the userdata script as a string literal in my YAML:

```yaml
      UserData:
        !Base64
          Fn::Sub: |
            #!/bin/bash -xe
            ...
```

But I'd like to be able to !Include it instead.

I was thinking of making a PR to add another type of include that wouldn't use the Fn::Join method, and instead just join the script with newlines into a single string. I'll put something together and let you know if it works.

Thanks for the great tool!

**Comment by monken** [comment]
Thank you very much for reporting and fixing this issue. I have made some changes:

* type: string will not support the `context` property. This makes the API more consistent as !Refs would not be supported which is really the primary use case for variables in includes. Fn::Sub supports custom variables out of the box so I think that should be the default.
* I deprecated type: literal altogether. The Fn::Sub syntax is superior and should be used going forward.

Let me know what you think and I will cut a release of you are happy!
